src/extends/room/territory.js

```diff
@@ -137,3 +139,340 @@ Room.prototype.getMineScore = function (roomName) {
+  score += getNeighborSourcesScore(roomName) * CITY_WEIGHTS_NEIGHBOR_SOURCES
+  score += getRegionDensityScore(roomName, 2) * CITY_WEIGHTS_REGION_DENSITY
+  score += getCityDistanceScore(roomName) * CITY_WEIGHT_CITY_DISTANCE
+  return score
+}
+
+Room.testRoomScore = function (roomName) {
+  const intel = Room.getIntel(roomName)
+  Logger.highlightData(intel)
+
+  Logger.highlight(`Two Sources: ${(intel[INTEL_SOURCES] - 1)}`)
+  Logger.highlight(`Two Sources: ${(intel[INTEL_SOURCES] - 1) * CITY_WEIGHTS_TWO_SOURCES}`)
+
+  Logger.highlight(`Walkability: ${intel[INTEL_WALKABILITY]}`)
+  Logger.highlight(`Walkability: ${intel[INTEL_WALKABILITY] * CITY_WEIGHTS_WALKABILITY}`)
+
+  Logger.highlight(`Swampiness: ${intel[INTEL_SWAMPINESS]}`)
+  Logger.highlight(`Swampiness: ${intel[INTEL_SWAMPINESS] * CITY_WEIGHTS_SWAMPINESS}`)
+
+  Logger.highlight(`Mineral Market: ${getMineralMarketScore(intel[INTEL_MINERAL])}`)
+  Logger.highlight(`Mineral Market: ${getMineralMarketScore(intel[INTEL_MINERAL]) * CITY_WEIGHTS_MINERAL_MARKET}`)
+
+  Logger.highlight(`Mineral Empire: ${getMineralEmpireNeedsScore(intel[INTEL_MINERAL])}`)
+  Logger.highlight(`Mineral Empire: ${getMineralEmpireNeedsScore(intel[INTEL_MINERAL]) * CITY_WEIGHTS_MINERAL_EMPIRE_NEED}`)
+
+  Logger.highlight(`Defensibility: ${getDefensibilityScore(roomName)}`)
+  Logger.highlight(`Defensibility: ${getDefensibilityScore(roomName) * CITY_WEIGHTS_DEFENSIBILITY}`)
+
+  Logger.highlight(`Room Type: ${getRoomTypeScore(roomName)}`)
+  Logger.highlight(`Room Type: ${getRoomTypeScore(roomName) * CITY_WEIGHTS_ROOMTYPE}`)
+
+  Logger.highlight(`Empire Clutter: ${getEmpireClutterScore(roomName)}`)
+  Logger.highlight(`Empire Clutter: ${getEmpireClutterScore(roomName) * CITY_WEIGHTS_EMPIRE_CLUTTERED}`)
+
+  Logger.highlight(`Empire Defense: ${getEmpireDefenseScore(roomName)}`)
+  Logger.highlight(`Empire Defense: ${getEmpireDefenseScore(roomName) * CITY_WEIGHTS_EMPIRE_DEFENSE}`)
+
+  Logger.highlight(`Region Resources: ${getRegionResourcesScore(roomName, 2)}`)
+  Logger.highlight(`Region Resources: ${getRegionResourcesScore(roomName, 2) * CITY_WEIGHTS_REGION_RESOURCES}`)
+
+  Logger.highlight(`Region Density: ${getRegionDensityScore(roomName, 2)}`)
+  Logger.highlight(`Region Density: ${getRegionDensityScore(roomName, 2) * CITY_WEIGHTS_REGION_DENSITY}`)
+
+  Logger.highlight(`City Distance: ${getCityDistanceScore(roomName)}`)
+  Logger.highlight(`City Distance: ${getCityDistanceScore(roomName) * CITY_WEIGHT_CITY_DISTANCE}`)
+
+  Logger.highlight(`Total Score: ${Room.getCityScore(roomName)}`)
+}
+
+function getMineralMarketScore (mineral) {
+  const max = qlib.market.getHighestMineralValue()
+  const cost = qlib.market.getAveragePrice(mineral, ORDER_SELL)
+  return cost / max
+}
+
+function getMineralEmpireNeedsScore (mineral) {
+  return 1 - Mineral.getEmpireRatio()[mineral]
+}
+
+function getDefensibilityScore (room) {
+  let score = 0
+  let totalEdge = 47 * 4
```

Comment:
if i made it 48 * 4 then it would be impossible for the room to get 100% because all rooms have some exits. That being said I don't feel strongly about this so if you want I'll change it.

Reply:
My bigger concern was that a completely open room could get a negative score, but I suppose that would be a good thing/working as intended.
As is, it can never get 100% anyway, as all rooms have some exits. Would probably need to use two different numbers on line 337 to make 100% possible.